### PR TITLE
[PLAT-306] Distinguish Logged in and Logged out users and content in GA

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Product wants to see if users are logged in when visiting certain pages.

## Changes

Google analytics js now gets a new dimension which checks if user is logged in. 

## QA Notes

No need for QA here.

## Side Effects

This will require additional configuration for our Google Analytics account which can be done by product like this: 

Find and create a custom dimension: 

screen shot 2018-01-29 at 11 43 35 am

Name it whatever you want.

screen shot 2018-01-29 at 11 44 33 am

Create a custom report with the pictured metrics and dimensions.

screen shot 2018-01-29 at 11 45 45 am

View the custom report with the secondary dimension of page destination.

screen shot 2018-01-29 at 11 46 54 am

## Ticket

https://openscience.atlassian.net/browse/PLAT-306

### CODE REVIEW

File: website/templates/project/project_base.mako [hunk 1]
```diff
@@ -1,5 +1,25 @@
 <%inherit file="../base.mako"/>
 
+<%def name="resource()"><%
+        if node.get('is_registration', False):
+           return 'registration'
+        elif node.get('is_preprint', False):
+           return 'preprint'
+        elif parent_exists:
+            return 'component'
```
Comment (pattisdr) [inline website/templates/project/project_base.mako]: Components show up as "projects".  My guess is that this method doesn't have access to parent_exists.  You might need to redefine here ` parent_exists = parent_node['exists']`.

File: website/templates/project/project_base.mako [hunk 2]
```diff
@@ -1,5 +1,25 @@
 <%inherit file="../base.mako"/>
 
+<%def name="resource()"><%
+        if node.get('is_registration', False):
+           return 'registration'
+        elif node.get('is_preprint', False):
+           return 'preprint'
```
Comment (pattisdr) [inline website/templates/project/project_base.mako]: Is this going to be confusing?  So the node that is associated with a preprint is being accessed here, but not the preprint itself.  For the actual preprint to send to GA that it is a preprint, that change would need to be made ember-side.

File: website/templates/project/project_base.mako [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+        elif parent_exists:
+            return 'component'
+        else:
+            return 'project'
+    %>
+</%def>
+
+<%def name="public()"><%
+        if node.get('is_public', False):
```
Comment (pattisdr) [inline website/templates/project/project_base.mako]: Haven't looked into why but public registrations still send is_public=False to GA.

File: website/templates/project/view_file.mako [hunk 4]
```diff
@@ -2,6 +2,28 @@
...
+            nodeType = 'preprint'
+        elif parent_exists:
+            nodeType = 'component'
+        else:
+            nodeType = 'project'
+        return 'file for a ' + nodeType
+%>
+</%def>
+
+<%def name="public()"><%
```
Comment (pattisdr) [inline website/templates/project/view_file.mako]: Can we define this `def` somewhere else and then just include it in these two files?

File: website/templates/project/view_file.mako [hunk 5]
```diff
@@ -2,6 +2,28 @@
 
 <div id="alertBar"></div>
 
+<%def name="resource()"><%
```
Comment (pattisdr) [inline website/templates/project/view_file.mako]: Is it possible to DRY this def up?   I see that the difference here is that you are adding the 'file for a ' to what you send to GA, unlike `project_base.mako` where you just send 'project' or something.  Not sure how hard it is to consolidate these defs - not a mako expert!

File: website/templates/project/project_base.mako (hunk 2 above)
Comment (Johnetordoff) [reply website/templates/project/project_base.mako]: I've talked to Project about this. They are okay with this for now, actual preprints that are to be covered by the Interfaces (I assume) and will will require some coordination at a later date.

File: website/templates/project/project_base.mako (hunk 3 above)
Comment (Johnetordoff) [reply website/templates/project/project_base.mako]: If your registration is still pending it remains private. Typically this happens until all contributors approve the registration or 48 hours elapse. I think this probably what's happening in your case, it caught me off guard as well.

**Comment by Johnetordoff** [comment]
Many excellent suggestion! Thanks @pattisdr!

File: website/templates/project/project_base.mako (hunk 3 above)
Comment (pattisdr) [reply website/templates/project/project_base.mako]: ahhh, right.

**Comment by pattisdr** [comment]
Changes look great @Johnetordoff, you can move to Passed CR when tests pass.

**Comment by Johnetordoff** [comment]
There's just an elasticsearch fail, passing this on.

File: website/templates/project/project_base.mako (hunk 3 above)
Comment (sloria) [reply website/templates/project/project_base.mako]: Minor: this can be simplified to `return node.get('is_public', False)`.

I can do this when I merge.

**Comment by sloria** [comment]
Merged into `release/next-platform` 🚢 🇮🇹

**Pull request closed without merging**